spinalcordtoolbox/scripts/sct_analyze_lesion.py

```diff
@@ -584,70 +595,91 @@ def __relative_ROIvol_in_mask(self, im_mask_data, im_atlas_roi_data, p_lst, im_t
+                                    desc="  Converting volume values into percentage values"):
+            idx = self.distrib_matrix_dct[sheet_name][self.distrib_matrix_dct[sheet_name].row == str(row)].index
             for tract_id in atlas_data:
                 val = self.distrib_matrix_dct[sheet_name].loc[idx, 'PAM50_' + str(tract_id).zfill(2)].values[0]
                 self.distrib_matrix_dct[sheet_name].loc[idx, 'PAM50_' + str(tract_id).zfill(2)] = val * 100.0 / vol_mask_tot
 
-    def __regroup_per_tracts(self, vol_dct, tract_limit):
-        res_mask = [vol_dct[t][0] for t in vol_dct if t >= tract_limit[0] and t <= tract_limit[1]]
-        res_tot = [vol_dct[t][1] for t in vol_dct if t >= tract_limit[0] and t <= tract_limit[1]]
+        # Add the total column
+        self.distrib_matrix_dct[sheet_name]['total % (all tracts)'] = \
+            self.distrib_matrix_dct[sheet_name].sum(numeric_only=True, axis=1)
+
+        # Add additional columns for the "CombinedLabels" defined by info_label.txt
+        for label_name, sublabels in self.atlas_combinedlabels.items():
+            column_names_to_sum = [f"PAM50_{str(subid).zfill(2)}" for subid in sublabels]
+            self.distrib_matrix_dct[sheet_name][label_name] = \
+                self.distrib_matrix_dct[sheet_name][column_names_to_sum].sum(axis=1)
+
+        # Add the total row
+        self.distrib_matrix_dct[sheet_name] = self.distrib_matrix_dct[sheet_name].append(
+            self.distrib_matrix_dct[sheet_name].sum(numeric_only=True, axis=0),
+            ignore_index=True
+        )
+        self.distrib_matrix_dct[sheet_name].iloc[
+            -1, self.distrib_matrix_dct[sheet_name].columns.get_loc('row')
+        ] = f'total % (all {self.row_name})'
+
+    def __regroup_per_tracts(self, vol_dct, tracts):
+        res_mask = [vol_dct[t][0] for t in vol_dct if t in tracts]
+        res_tot = [vol_dct[t][1] for t in vol_dct if t in tracts]
         return np.sum(res_mask) * 100.0 / np.sum(res_tot)
 
     def _measure_totLesion_distribution(self, im_lesion, atlas_data, im_vert, p_lst):
 
         sheet_name = 'ROI_occupied_by_lesion'
-        self.distrib_matrix_dct[sheet_name] = pd.DataFrame.from_dict({'vert': [str(v) for v in self.vert_lst] + ['total % (all vert)']})
+        total_row = f'total % (all {self.row_name})'
+        rows_with_total = {
+            **self.rows,
+            # numpy array index equivalent to [:, :, :]
+            total_row: (slice(None), slice(None), slice(None)),
+        }
+        self.distrib_matrix_dct[sheet_name] = pd.DataFrame.from_dict({'row': [str(r) for r in rows_with_total]})
 
         # initialized to 0 for each vertebral level and each PAM50 tract
         for tract_id in atlas_data:
-            self.distrib_matrix_dct[sheet_name]['PAM50_' + str(tract_id).zfill(2)] = [0] * len(self.vert_lst + ['total % (all vert)'])
+            self.distrib_matrix_dct[sheet_name]['PAM50_' + str(tract_id).zfill(2)] = [0] * len(rows_with_total)
```

Comment:
To align with the code above:
```suggestion
            self.distrib_matrix_dct[sheet_name][f"PAM50_{tract_id:02}"] = [0] * len(rows_with_total)
```